### DISCUSSION TITLE
Issues logging with comet

### DISCUSSION THREAD

**Discussion by robmarkcole** [ultralytics/yolov5#9840] [Q&A] [answered]
The integration with [comet](https://github.com/ultralytics/yolov5/tree/master/utils/loggers/comet) looks neat I have tried all 3 approaches without success:

- Setting API key as env var
- Creating config file
- Using the python API (below)

```python3
import comet_ml
comet_ml.init(project_name='mine', api_key='mine')

... 
COMET INFO: Comet API key is valid
```

The user guide states `Comet will automatically log your hyperparameters` - but this is in conflict with the example notebook which states that `Logging Models to Comet is disabled by default. To enable it, pass the save-period argument to the training script.`

Possibly this is due to old version of yolov5 in the Roboflow notebook?

```
# clone YOLOv5 repository
!git clone https://github.com/ultralytics/yolov5  # clone repo
%cd yolov5
!git reset --hard fbe67e465375231474a2ad80a4389efc77ecff99
```

Just an additional comment - what happened to support for wandb?

**Comment by glenn-jocher** [comment]
@robmarkcole thanks for the feedback!

@DN6 could you take a look at this please?

**Comment by DN6** [comment, marked as answer]
Hi @robmarkcole could you share some more details on where you're trying to set your Comet credentials? (Colab Notebook, local machine, a server etc)

Regarding the user guide and model logging. The Comet integration automatically logs hyperparameters, metrics, images (PR Curve, Confusion Matrix, etc). Model Checkpoints are not logged automatically by default. But if you pass the `save-period` command line argument to the `train.py` script, the model checkpoints will be logged to Comet.

**Reply by nsriniva03** [reply]
According to the documentation, we should be able to log the model predictions as shown in the example project (https://www.comet.com/examples/comet-example-yolov5/view/1c4Dqcu8mZ767NBipjwlx3gz6/panels?shareable=YcwMiJaZSXfcEXpGOHDD12vA1&utm_source=yolov5&utm_medium=partner&utm_campaign=partner_yolov5_2022&utm_content=comet_colab)

However, when I run the following command,
 python train.py  --img 640 --batch 16 --epochs 5 --data coco128.yaml --weights yolov5s.pt --bbox_interval 1

The bounding box panel does not get logged and I am unable to view the model predictions. Any suggestions or thoughts on how to resolve the issue?

Thank you
